floyd/cli/experiment.py

```diff
@@ -150,19 +150,22 @@ def logs(id, url, tail, follow, sleep_duration=1):
     Print the logs of the run.
     """
     tail = tail or follow
-    try:
-        experiment = ExperimentClient().get(normalize_job_name(id))
-    except FloydException:
-        experiment = ExperimentClient().get(id)
 
-    if experiment.state == 'queued':
-        floyd_logger.info("Job is currently in a queue")
-        return
+    log_msg_printed = False
+    while True:
+        try:
+            experiment = ExperimentClient().get(normalize_job_name(id))
+        except FloydException:
+            experiment = ExperimentClient().get(id)
 
-    instance_log_id = experiment.instance_log_id
-    if not instance_log_id:
-        floyd_logger.info("Job not started yet, no log to show.")
-        sys.exit(1)
+        instance_log_id = experiment.instance_log_id
+        if instance_log_id:
+            break
+        elif not log_msg_printed:
+            floyd_logger.info("Waiting for logs ...\n")
+            log_msg_printed = True
+
+        sleep(0.2)
```

Comment:
Everything else looks good to me, except hitting an API every 0.2 second is too frequent :P I would prefer it to be set to 1-3 second range.

We will be releasing an event based realtime API soon, so a lot of the pulling code can be dropped in the future ;)